backend/infrahub/core/attribute.py

```diff
@@ -915,6 +916,107 @@ def to_db(self) -> dict[str, Any]:
         return data
 
 
+class MacAddress(BaseAttribute):
+    type = str
+
+    @property
+    def obj(self) -> netaddr.EUI:
+        """Return the MAC adress."""
+        if not self.value:
+            raise ValueError("value for MAC address must be defined")
+        return netaddr.EUI(addr=self.value)
+
+    @property
+    def oui(self) -> Optional[str]:
+        """Return the OUI (Organisationally Unique Identifier) for the MAC address."""
+        if not self.value:
+            return None
+        try:
+            return str(self.obj.oui)
+        except netaddr.NotRegisteredError:
+            # Workaround OUI lookup failure
+            # netaddr might not be up-to-date, or actual OUI just not exist (i.e. random mac address)
+            return str(self.obj).removesuffix(f"-{self.ei}")
+
+    @property
+    def ei(self) -> Optional[str]:
+        """Return the EI (Extension Identifier) for the MAC address."""
+        if not self.value:
+            return None
```

Comment:
As we have many "if not self.value:" it would be nice with a test where the mac address was not found just to catch these lines in the coverage too.